modules/global/src/com/haulmont/chile/core/model/utils/MethodsCache.java

```diff
@@ -18,59 +18,103 @@
 
 import org.apache.commons.lang3.StringUtils;
 
+import java.lang.invoke.*;
 import java.lang.reflect.Method;
-import java.lang.reflect.InvocationTargetException;
-import java.util.Map;
 import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
 
 public class MethodsCache {
 
-    private final transient Map<String, Method> getters = new HashMap<>();
-    private final transient Map<String, Method> setters = new HashMap<>();
+    private final transient Map<String, Function> getters = new HashMap<>();
```

Comment:
I think we can remove transient from fields, MethodsCache is not Serializable anyway.